backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchStringIndexFieldTypeOptionsStep.java

```diff
@@ -231,12 +252,58 @@ private boolean resolveNorms() {
 	}
 
 	private String resolveTermVector() {
-		switch ( termVector ) {
-			case NO:
-			case DEFAULT:
-				return "no";
-			default:
-				return termVector.name().toLowerCase( Locale.ROOT );
+		if ( highlightable.contains( Highlightable.FAST_VECTOR ) || highlightable.contains( Highlightable.ANY ) ) {
+			switch ( termVector ) {
+				case WITH_POSITIONS_PAYLOADS:
+				case WITH_POSITIONS_OFFSETS_PAYLOADS:
+					return TermVector.WITH_POSITIONS_OFFSETS_PAYLOADS.name().toLowerCase( Locale.ROOT );
+				default:
+					return TermVector.WITH_POSITIONS_OFFSETS.name().toLowerCase( Locale.ROOT );
+			}
 		}
+		else {
+			switch ( termVector ) {
+				case NO:
+				case DEFAULT:
+					return "no";
+				default:
+					return termVector.name().toLowerCase( Locale.ROOT );
+			}
+		}
+	}
+
+	private Set<SearchHighlighterType> resolveAllowedHighlighterTypes() {
```

Comment:
the `DEFAULT` was checked as part of  `FullTextFieldProcessor` logic... but since we have some much already happening here ... I'll add a check here and clean up the `FullTextFieldProcessor`

Reply:
Careful, not all fields are configured through `FullTextFieldProcessor`. There's programmatic mapping, and then there's also field definition within bridges. Yes, you definitely need to assume you can get a `DEFAULT` here.